### DISCUSSION TITLE
Differences between GPUTexture and GPUTextureView?

### DISCUSSION THREAD

**Discussion by wewindy** [gpuweb/gpuweb#2641] [Q&A] [answered]
Why not just use `GPUTexture` as `WebGLTexture`? I wonder the purpose to use view of texture object.

**Comment by Kangz** [comment, marked as answer]
In WebGL `WebGLTexture` contains both the content of the texture as well as some mutable state, like the array layer range, or mip level range selected. This means that it is impossible to the same `WebGLTexture` with different ranges in the same drawcall. WebGPU separates the content of the texture (`GPUTexture`) and the mutable state of the texture (`GPUTextureView` and `GPUSampler`).

**Reply by wewindy** [reply]
Thank you!
